### PR TITLE
Add `remove-node` recipe

### PR DESCRIPTION
**Pull request by jletey**
This is my initial start for the "Remove a node" recipe. It's pretty much just a copy of @Murderlon's ["Find a node"](https://unifiedjs.com/learn/recipe/find-node/), but updated to use the [`unist-util-remove`](https://github.com/syntax-tree/unist-util-remove) package. Open to suggestions - I'm not 💯 certain that this is the best way to depict how to use the package, but you guys are the experts 🙌

### CODE REVIEW

File: doc/learn/remove-node.md [hunk 1]
```diff
@@ -1,15 +1,118 @@
...
+### Contents
 
-[github]: https://github.com/unifiedjs/unifiedjs.github.io/blob/src/doc/learn/remove-node.md
+*   [What is a node](#what-is-a-node)
+*   [Removing a node](#removing-a-node)
+
+### What is a node
+
+A node is a single language specific unit inside a [syntax tree][syntax-tree].
+For example: a heading in markdown, or anchor element in HTML. In unified, nodes
```
Comment (wooorm) [inline doc/learn/remove-node.md]:
```suggestion
For example: a heading in markdown, or anchor element in HTML.  In unified, nodes
```

File: doc/learn/remove-node.md [hunk 2]
```diff
@@ -1,15 +1,118 @@
...
+follow the [unist][] specification.
+
+### Removing a node
+
+The concept of removing a node involves
+[tree traversal][tree-traversal] of a [syntax tree][syntax-tree].
+
+Unified compatible utilities should be used for finding a node.
+Utilities are functions that work with nodes. All specifications
```
Comment (wooorm) [inline doc/learn/remove-node.md]:
```suggestion
Utilities are functions that work with nodes.  All specifications
```

File: doc/learn/remove-node.md [hunk 3]
```diff
@@ -1,15 +1,118 @@
...
+that extend [unist][] can use the [unist utilities][unist-utils],
+but they can also have their own utilities for more specific nodes.
+
+To start removing nodes for your input you'll need:
```
Comment (wooorm) [inline doc/learn/remove-node.md]:
```suggestion
To start removing nodes for your input you’ll need:
```

File: doc/learn/remove-node.md [hunk 4]
```diff
@@ -1,15 +1,118 @@
...
+
+*   A processor (such as [`remark`][remark]).
+*   A utility of choice.
+
+For this example we use [`remark`][remark]
+and [`unist-util-remove`][unist-util-remove]. We want to remove
```
Comment (wooorm) [inline doc/learn/remove-node.md]:
```suggestion
and [`unist-util-remove`][unist-util-remove].  We want to remove
```

File: doc/learn/remove-node.md [hunk 5]
```diff
@@ -1,15 +1,118 @@
...
-description: How to remove nodes in a tree
-published: 2019-12-12
-modified: 2019-12-12
+description: How to remove a node in any unified flavored syntax tree.
+tags:
+  - node
+  - tree
+  - traverse
+  - walk
+  - remove
```
Comment (wooorm) [inline doc/learn/remove-node.md:10]:
```suggestion
  - remove
  - delete
```

**Review by wooorm — COMMENTED** [review]
This works!

- I think it would be good to show different processors in different recipes. Not just remark, but for example rehype
- Removing nodes is a more complex that finding nodes.
  In this example, users may want to replace the node with its contents, so it may be interesting to add how to do that, pseudocode:

  ```js
  visit(tree, function (node, index, parent) {
    if (node.type === 'emphasis') {
      // Replace the node with its contents
      parent.children.splice(index, 1, node.children)
      // Don’t visit `node`, instead, continue walking the tree at the node *now*
      // at index:
      return [visit.SKIP, index]
    }
  })
  ```

**Comment by wooorm** [comment]
What do you think of my review comments?

**Comment by jletey** [comment]
@wooorm Sorry for going radio silent. Very rude of me 😬

> * I think it would be good to show different processors in different recipes. Not just remark, but for example rehype

Okay - so have a section explaining how to do this with remark and another for rehype? Should I update the ["Find a node"](https://unifiedjs.com/learn/recipe/find-node/) recipe as well?

> * Removing nodes is a more complex that finding nodes.
>   In this example, users may want to replace the node with its contents, so it may be interesting to add how to do that, pseudocode:
>   ```js
>   visit(tree, function (node, index, parent) {
>     if (node.type === 'emphasis') {
>       // Replace the node with its contents
>       parent.children.splice(index, 1, node.children)
>       // Don’t visit `node`, instead, continue walking the tree at the node *now*
>       // at index:
>       return [visit.SKIP, index]
>     }
>   })
>   ```

I agree - this is something worthwhile to add 👍

**Comment by wooorm** [comment]
Re remark & rehype: I was thinking more switching them out per recipe, so rehype for this one? Remark for the find node?

**Comment by jletey** [comment]
> I was thinking more switching them out per recipe, so rehype for this one? Remark for the find node?

@wooorm Hmm, not a bad idea - but what exactly are the use cases for the recipes? Let's say I'm a user that wants to learn both methods but solely for remark (or solely for rehype) - but each recipe is using a different processor.

**Comment by wooorm** [comment]
I think recipes should always be vague enough for you to understand how to make this complete dish, and not delve into steps, e.g., how to best dice an onion; that should go in a different recipe.

Wanting to remove a node in remark is an XY problem: you think that's what you want, but there's a failure to understand unified; the actual problems are a) how to access the remark AST, and b) how to remove a node in unified

**Comment by jletey** [comment]
Thank you @wooorm for going more in depth about this 🙏 I'll use rehype for this recipe and also include the pseudocode for replacing a node with it's children.

**Comment by wooorm** [comment]
@johnletey friendly ping! 🛎 Hope you’re safe!

**Comment by johnletey** [comment]
@woorm Hey :wave: Unfortunately I’ve been locked out of my old account :sweat_smile:

I don’t know the best way to go about completing this PR, as neither me nor GitHub Support can regain access to my old account.

Should I just go ahead and create a new PR?

Hope you are well in these trying times!

**Comment by wooorm** [comment]
Oh that sounds quite unfortunate! Hmm, New PR would work, or I can help fill the gaps too, whatever works for you!

**Comment by johnletey** [comment]
If you don’t mind, let’s do a new PR, but would love some help!

**Comment by wooorm** [comment]
Whoops, I just wrote and pushed a `remove-node` recipe but forgot about this pr 😥

**Comment by johnletey** [comment]
@wooorm No problem at all

**Pull request closed without merging**